test/adapters/level_zero/CMakeLists.txt

```diff
@@ -45,6 +45,7 @@ if(UR_BUILD_ADAPTER_L0)
         # Make L0 use CallMap from a seprate shared lib so that we can access the map
         # from the tests. This only seems to work on linux
         add_library(zeCallMap SHARED zeCallMap.cpp)
+        install_ur_library(zeCallMap)
```

Comment:
why are we installing this?

Reply:
We could think about some other alternative to creating zeCallMap library. Perhpas we could just define a public function in Level Zero adapter that would return the pointer to zeCallMap (only if UR_L0_CALL_COUNT_IN_TESTS  is defined)?

Reply:
@igchor I agree with @lslusarczyk. This is a horrible hack that needs to removed. The L0 adapter library has a dependency on this test-only object (!!).

I really don't understand what we are trying to test. Piling more hacks on top of this doesn't seem like a good idea. We need to rethink this approach.

Reply:
> The tests will compile fine but you will get failure on dlopen.

Indeed. I get error on dlopen.

> We need to rethink this approach.

It seems you've already designed a solution here: https://github.com/oneapi-src/unified-runtime/issues/1454

I think I should neither implement it as a part of this PR nor block this PR because we lack of clean solution. I'm reverting to my original install solution that installs zeCallMap library if level_zero_adapter was compiled with UR_L0_CALL_COUNT_IN_TESTS define. It is needed to be installed because otherwise installed libur_adapter_level_zero.so can not be loaded. I'll add a note to revert this when https://github.com/oneapi-src/unified-runtime/issues/1454 is done.

Good point is that hack is now more visible, "stings the eyes" more, so we will sooner fix it, I think.